include/merlin/core_kernels/kernel_utils.cuh

```diff
@@ -32,41 +32,68 @@ namespace cg = cooperative_groups;
 namespace nv {
 namespace merlin {
 
-// if i % 2 == 0, select buffer 0, else buffer 1
+// Vector Type of digests for memory access.
+using VecD_Load = byte16;
+// Vector Type of digests for computation.
+using VecD_Comp = byte4;
+
+// Select from double buffer.
+// If i % 2 == 0, select buffer 0, else buffer 1.
 __forceinline__ __device__ int same_buf(int i) { return (i & 0x01) ^ 0; }
-// if i % 2 == 0, select buffer 1, else buffer 0
+// If i % 2 == 0, select buffer 1, else buffer 0.
 __forceinline__ __device__ int diff_buf(int i) { return (i & 0x01) ^ 1; }
 
 template <typename K>
-__forceinline__ __device__ uint8_t empty_digest() {
+__forceinline__ __host__ __device__ D empty_digest() {
```

Comment:
I can see you have called the function in any host code, if I'm right , the `__host__` is unnecessary.

Reply:
Yes, there is no calling to this function in host side at present.
I just want to emphasize this function can be called on both host and device side.
OK, I think I will remove the `__host__`, as the `digests` is not visible to users and only be operated on device side.